src/ui/src/builder/useComponentDescription.ts

```diff
@@ -18,6 +18,21 @@ export function getSourceBlueprintName(
 	return sourceBlueprint?.content?.key || null;
 }
 
+/**
+ * Gets the description for a shared blueprint by looking up its source blueprint.
+ * Returns the source blueprint's deployedDescription, or null if not found.
+ */
+export function getSourceBlueprintDescription(
+	wf: Core,
+	component: Component,
+): string | null {
+	if (component?.type !== "blueprints_shared") return null;
+	const sourceBlueprintId = component.content?.sourceBlueprintId;
+	if (!sourceBlueprintId) return null;
+	const sourceBlueprint = wf.getComponentById(sourceBlueprintId);
+	return sourceBlueprint?.content?.deployedDescription || null;
+}
```

Comment:
I feel we most often use `undefined`
```suggestion
): string | undefined {
	if (component?.type !== "blueprints_shared") return;
	const sourceBlueprintId = component.content?.sourceBlueprintId;
	if (!sourceBlueprintId) return;
	const sourceBlueprint = wf.getComponentById(sourceBlueprintId);
	return sourceBlueprint?.content?.deployedDescription;
}
```

Reply:
@madeindjs updated